File: src/entities/Restaurant.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  OneToOne,
  JoinColumn,
  OneToMany,
} from "typeorm";
import { User } from "./User";
import { MenuItem } from "./MenuItem";
import { Order } from "./Order";

@Entity("restaurants")
export class Restaurant {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column()
  userId!: string;

  @OneToOne(() => User, (user) => user.restaurant)
  @JoinColumn({ name: "userId" })
  user!: User;

  @Column()
  name!: string;

  @Column({ nullable: true })
  description?: string;

  @Column()
  address!: string;

  @OneToMany(() => MenuItem, (item) => item.restaurant)
  menuItems!: MenuItem[];

  @OneToMany(() => Order, (order) => order.restaurant)
  orders!: Order[];
}

File: flutter_app/lib/core/models/restaurant.dart
import 'menu_item.dart';

class Restaurant {
  final String id;
  final String userId;
  final String name;
  final String? description;
  final String address;
  final List<MenuItem> items;

  const Restaurant({
    required this.id,
    required this.userId,
    required this.name,
    this.description,
    required this.address,
    this.items = const [],
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) => Restaurant(
    id: json['id'] as String,
    userId: json['userId'] as String,
    name: json['name'] as String,
    description: json['description'] as String?,
    address: json['address'] as String,
    items: (json['items'] as List<dynamic>?)
        ?.map((e) => MenuItem.fromJson(e as Map<String, dynamic>))
        .toList() ?? [],
  );
}
